transcom/mymove
# [MB-2622] TIO navigate between payment request service items

## Description

After a payment request is available from the prime, the TIO can review (approve or reject) all service items from the associated shipments (hhg, nts) and misc service items (counseling fee, shipment management fee).

This component is the sidebar to the document viewer that will display the uploaded payment request pdf or images.  In this PR we order the service items by createdAt and add the footer UI to cycle between the service items and add a placeholder for the sum total of approved amounts.

## Reviewer Notes

I implemented a sorting of the service items by shipment group, where it takes the earliest createdAt date of each service item in each unique shipment and uses that to order the groups.  I think this is the behavior we want because it would be confusing to the user if you were approving or rejecting service items randomly outside of their shipment group.

We could easily adjust this to always put the Basic Service Items with no shipment (Counseling Services and Shipping Management fees) to always come first like it looks like later mocks may want (or stacked on the same screen).

This is something to consider once the data loading for the payment requests index page (moves/:id/payment-requests) becomes clearer.

I noticed too that we should start getting consistent around labels for service items so we aren't calling them different things on the approve MTO form, in the database, and then in the review.

Here are the current database codes and labels:
![Screen Shot 2020-07-14 at 5 32 40 PM](https://user-images.githubusercontent.com/52669884/87478611-113d1b00-c5f8-11ea-9499-6a041922f8df.png)

## Setup

```shell
generate-test-data --named-scenario e2e_basic
make server_run
make office_client_run
```

1. Login as a TIO user to the office app
2. Navigate to the Payment request review URL `/moves/1/payment-requests/a2c34dba-015f-4f96-a38b-0c0b9272e208`
3. The Previous button should be disabled on the first service item for review
4. Use the next button to cycle through the service items
5. At the last item the Next button is set to disabled
6. Clicking previous will let you navigate in reverse order back to the beginning

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/backend#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2622) for this change
* [Storybook component](http://localhost:6006/?path=/story/too-tio-components-review-service-items)
* [Abstract Mock](https://share.goabstract.com/3cff868b-be81-401c-82fc-3634b1e26b52?collectionLayerId=80f03f6d-4eb8-46ef-908c-ff46af5a525c&mode=design)

## Screenshots

![review-service-items-navigate](https://user-images.githubusercontent.com/52669884/87721421-673ec980-c784-11ea-8b20-535a55a72e30.gif)
